script.js

```diff
@@ -1,12 +1,363 @@
+async function fetchAPI(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Could not fetch data from ${url}`);
+  }
+  return response.json();
+}
+
+function addCardLayout() {
+  layoutSelector.classList.add("layout__for__cards");
+  searchField.classList.add("test");
+}
+
+function removeCardLayout() {
+  layoutSelector.classList.remove("layout__for__cards");
+}
+
+// Fetches shows data via API
+async function getAllShows() {
+  try {
+    if (!currentState.showsFetched) {
+      currentState.allShows = await fetchAPI("https://api.tvmaze.com/shows");
+      currentState.showsFetched = true;
+    }
+    return currentState.allShows;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function getAllEpisodes() {
+  try {
+    if (!currentState.episodesFetched) {
+      currentState.allEpisodes = await fetchAPI(
+        `https://api.tvmaze.com/shows/${currentState.showsID}/episodes`
+      );
+      currentState.episodesFetched = true;
+    }
+    return currentState.allEpisodes;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+// // Creating cards
+
+// // Creates element and adds class if required
+function createElement(tag, className) {
+  const element = document.createElement(tag);
+  if (className) {
+    element.classList.add(className);
+  }
+  return element;
+}
+
+// Adds padding to the start of season and episode
+// Example --- S1E4 = S01E04
+function formatSeasonAndEpisode(season, episodeNumber) {
+  return `S${String(season).padStart(2, "0")}E${String(episodeNumber).padStart(
+    2,
+    "0"
+  )}`;
+}
+
+// Creates card component
+function createShowCard(show) {
+  const cardDiv = createElement("section", "card");
+
+  const layoutSelector = document.querySelector(".layout");
+
+  const seasonAndEpisodeWithPadding = formatSeasonAndEpisode(
+    show.season,
+    show.number
+  );
+
+  const filmTitleElement = createElement("h1", "card__title");
+  filmTitleElement.textContent = `${show.name}`;
+  if (show.season) {
+    filmTitleElement.textContent = `${show.name} - ${seasonAndEpisodeWithPadding}`;
+  }
+  cardDiv.appendChild(filmTitleElement);
+
+  const wrapImageAndSummary = createElement("div", "img__summary");
+
+  const filmImgElement = createElement("img", "card__img");
+  filmImgElement.setAttribute("src", show.image.original);
+  wrapImageAndSummary.appendChild(filmImgElement);
+
+  const filmSummaryElement = createElement("p", "card__summary");
+  filmSummaryElement.innerHTML = show.summary;
+  wrapImageAndSummary.appendChild(filmSummaryElement);
+  cardDiv.appendChild(wrapImageAndSummary);
+
+  const additionalDetails = createElement("div", "additionaldetails");
+  const rating = createElement("p", "additionaldetails__text");
+  rating.innerHTML = `Rating: ${show.rating.average} <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-6 h-6 star__rating">
```

Comment:
Since the [⭐] icon is for cosmetic purposes. How could you apply this using css instead?

**Hint** - `after` pseudo element